core/embed/rust/src/ui/layout/result.rs

```diff
@@ -0,0 +1,34 @@
+use crate::micropython::{
+    obj::{Obj, ObjBase},
+    qstr::Qstr,
+    typ::Type,
+};
+
+#[repr(C)]
+pub struct ResultObj {
+    base: ObjBase,
+}
+
+impl ResultObj {
+    /// Convert ResultObj to a MicroPython object.
+    pub const fn as_obj(&'static self) -> Obj {
+        // SAFETY:
+        //  - We are an object struct with a base and a type.
+        //  - 'static lifetime holds us in place.
+        //  - There's nothing to mutate.
+        unsafe { Obj::from_ptr(self as *const _ as *mut _) }
+    }
+}
+
+// SAFETY: We are in a single-threaded environment.
+unsafe impl Sync for ResultObj {}
+
+static CONFIRMED_TYPE: Type = obj_type! { name: Qstr::MP_QSTR_CONFIRMED, };
+static CANCELLED_TYPE: Type = obj_type! { name: Qstr::MP_QSTR_CANCELLED, };
```

Comment:
Implemented this as two distinct types because it was the least comlex way to make python `print()` the value. Is there a reason not to do it like this.

Reply:
It takes a bit more flash but I wouldn't sweat it too much.

Reply:
For a smaller version, we could just make two `mp_obj_object_t`s. The option to print the object name wouldn't be there but we don't _really_ need that.

But this is good enough I think.

Reply:
Ok, I'll leave it like this now and rewrite it once we add more result constants.